tests/rptest/tests/transactions_test.py

```diff
@@ -260,6 +260,44 @@ def change_static_member_test(self):
 
 
 class UpgradeTransactionTest(RedpandaTest):
+    @cluster(num_nodes=2, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def consume(self, consumer, max_records=10, timeout_s=2):
+        def consume_records():
+            records = consumer.consume(max_records, timeout_s)
+
+            if (records != None) and (len(records) != 0):
+                return True, records
+            else:
+                False
+
+        return wait_until_result(consume_records,
+                                 timeout_sec=30,
+                                 backoff_sec=2,
+                                 err_msg="Can not consume data")
+
+    def check_consume(self, topic_name, max_tx):
+        consumer = ck.Consumer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'group.id': f"consumer-{uuid.uuid4()}",
+            'auto.offset.reset': 'earliest',
+        })
+
+        consumer.subscribe([topic_name])
+
+        num_consumed = 0
+        prev_rec = bytes("0", 'UTF-8')
+
+        while num_consumed < max_tx:
```

Comment:
This will still loop forever if anything goes wrong with the consume: this loop should have a timeout check and throw an exception if too much time has passed and the condition still hasn't been satisfied.

Reply:
No
1) We have until_wait for consume command
2) We have check for consumed record after it

So we can not consume empty records forever
and will check them